### PR TITLE
Refactor api tests for modular pipelines change

### LINKED ISSUE #1899: Refactor Modular Pipelines in the Kedro-viz backend
## Description

The logic of how we calculate modular pipelines in Kedro-viz backend differs from how it is computed in Kedro. For instance, modular pipelines are not associated with namespaces, and neither are the datasets in a modular pipeline. Only nodes have namespaces, and based on the way the datasets are consumed by different nodes, these datasets can be regarded as input/output/children to each modular pipeline.

This PR is to revise the current logic and make it more like the way Kedro does it. Modular Pipelines have caused a lot of issues for the user based on the difference in approach. With this refactor, we hope to align the approaches.

## Context
Why is this change important to you? How would you use it? How can it benefit other users?

## Possible Implementation
(Optional) Suggest an idea for implementing the addition or change.

## Possible Alternatives
(Optional) Describe any alternative solutions or features you've considered.

## Checklist

- [ ] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Partially resolves #1899 

## Development notes
* Updated cypress tests as the menu now shows the inputs/outputs datasets for a focussed modular pipeline
* Update conftest fixtures with edges case pipeline scenarios mentioned in -
    - https://github.com/kedro-org/kedro-viz/pull/1651
    - https://github.com/kedro-org/kedro-viz/issues/1814
* Updated REST API responses as the modular pipeline mapping changed 

## QA notes
- This PR is part of a bigger refactor (https://github.com/kedro-org/kedro-viz/pull/1897) of modular pipelines and is created to ease review process.
- The CI build might fail as this PR is not self-sufficient

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: package/tests/conftest.py [hunk 1]
```diff
@@ -87,6 +91,207 @@ def train_model(model_inputs, parameters):
...
+                func=lambda x: x,
+                inputs="input_df",
+                outputs="output_df",
+                name="generic_node",
+            ),
+        ]
+    )
+
+
+# [TODO: Need to better manage this fixture]
```
Comment (merelcht) [inline package/tests/conftest.py:108]: I see there's still a TODO here. The fixture is really large and quite hard to read IMO. Can we perhaps come up with example pipelines with more self-explanatory names and variable names? `dataset_1` etc is a bit hard to follow.. Or at least can we add more explicit notes that explain what each of these use_cases are for?

Comment (ravi-kumar-pilla) [reply package/tests/conftest.py:108]: Hi @merelcht , Yes even I felt the same. So for now kept a TODO and added links in the docstring of the fixture so the devs can take a look. I wanted to get some help from reviewers on how to do this more efficiently.

Based on your suggestion, I will try to come up with more meaningful names and see if we can split the fixture into multiple parts for each use case. Thank you

**Review by merelcht — APPROVED** [review]
This is a huge improvement! The test fixtures are much easier to understand now. Thanks @ravi-kumar-pilla ⭐ ⭐ ⭐ !
